docs/software_requirements/system_initialization.sdoc

```diff
@@ -10,3 +10,269 @@ IMPORT_FROM_FILE: software_requirements.sgra
 TBD
 
 [/FREETEXT]
+
+[SECTION]
+TITLE: Interface Requirements
+
+[FREETEXT]
+
+TBD
+
+[/FREETEXT]
+
+[REQUIREMENT]
+UID: ZEP-127
+STATUS: Draft
+TYPE: Interface
+COMPONENT: System Initialization
+TITLE: Build-time Registration for System Initialization Notifications
+STATEMENT: >>>
+The Zephyr RTOS shall expose an interface to build-time registration for the following system initialization notifications:
+
+- Pre-CPU Architecture Initialization
+- Pre-Kernel Initialization
+- Post-Kernel Initialized
+- Application Initialization
+<<<
```

Comment:
This needs a notification that System Initialization is compete. This notification needs to be given immediately before the call to `main(void)`.

This event is needed by components that change behavior upon initialization complete (e.g., user interfaces, allowing communication). Providing this through the System Initialization component allows independence from the implementation of `main()` (whether defined by the test framework or the default implementation in the kernel).